### PR TITLE
feat: results screen

### PR DESCRIPTION
**Pull request by Magdaline-A**
# DCMAW-7210: Success screen

Added a success/ results screen 

# Checklist

## Before raising your pull request:
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with ticket ID and a short description about the feature or update
      i.e. _DCMAW-222: Added ReadID SDK to iOS app_
- [ ] Created a `draft` pull request if it is not yet ready for review

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

- [ ] Met all accessibility requirements?
    - [ ] Checked dynamic type sizes are applied
    - [ ] Checked VoiceOver can navigate your new code
    - [ ] Checked a user can navigate only using a keyboard around your new code 

## Before merging your pull request:
- [ ] Ensure that the code coverage and SonarCloud checks have passed
- [ ] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [ ] Ran the app to ensure that no regressions have been caused by changes during code review.
- [ ] Targeted the correct branch; `develop`, `release` or `main`

### CODE REVIEW

File: Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/Results/ResultsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import UIKit
+
+/// View controller for `ResultsView` screen.
+public final class ResultsViewController: UIViewController {
```
Comment (robinphillips) [inline Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/Results/ResultsViewController.swift:4]: As you highlighted this morning, we need to decide whether to figure out the merge order for this or https://github.com/govuk-one-login/mobile-ios-common/pull/27 

If the other `BaseViewController` PR comes first, then this PR should likely be updated to inherit from `BaseViewController` and the view model protocol would need updating too. Happy to catch up about that if / when it happens if that would be helpful.

File: Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/Results/ResultsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public override var nibName: String? { "ResultsView" }
+    
+    public let viewModel: ResultsViewModel
+
+    public init(viewModel: ResultsViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: "ResultsView", bundle: .module)
+    }
+
```
Comment (robinphillips) [inline Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/Results/ResultsViewController.swift:13]: Subject to the comment above about merge order of PRs, if this comes first:
- Could this screen ever have a back button? If yes, then it should have the back button method for setting the label
- It should probably have the right bar button methods / dismiss method for setting dismiss button if it is displayed modally

If this PR is merged after the `BaseViewController` PR, then that functionality would be included for 'free' so maybe wait to see :)

Comment (Magdaline-A) [reply Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/Results/ResultsViewController.swift:13]: For the page I'm making this for in Wallet, the intention is that the main button (`resultsButton`) will dismiss the page. But yes, I agree that for future use we may need a back button :)

**Review by MarieseMikely — APPROVED** [review]

File: Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/Results/ResultsViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import UIKit
+
+/// Protocol for the view model required to initilise a ``ResultsViewController``
+public protocol ResultsViewModel {
+    var image: String { get }
+    var title: GDSLocalisedString { get }
+    var body: GDSLocalisedString? { get }
```
Comment (robinphillips) [inline Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/Results/ResultsViewModel.swift:7]: I realise this PR has now already been merged...

Is there a reason for the `body` property being optional @Magdaline-A ?

**Pull request merged**